src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -302,28 +302,36 @@ internal void ParseUsingNewParser()
         {
             ISolutionSerializer serializer = SolutionSerializers.GetSerializerByMoniker(FullPath);
 
-            if (serializer != null)
+            if (serializer == null)
+            {
+                ProjectFileErrorUtilities.ThrowInvalidProjectFile(
+                    new BuildEventFileInfo(FullPath),
+                    $"InvalidProjectFile",
+                    $"No solution serializer was found for {FullPath}");
+            }
+            else
             {
                 try
                 {
-                    SolutionModel solutionModel = serializer.OpenAsync(FullPath, CancellationToken.None).Result;
+                    SolutionModel solutionModel = serializer.OpenAsync(FullPath, CancellationToken.None).GetAwaiter().GetResult();
                     ReadSolutionModel(solutionModel);
                 }
+                catch (SolutionException solutionEx)
+                {
+                    var errorLocation = ElementLocation.Create(FullPath, solutionEx.Line ?? 0, solutionEx.Column ?? 0);
+                    ProjectFileErrorUtilities.ThrowInvalidProjectFile(
+                        new BuildEventFileInfo(errorLocation),
+                        "InvalidProjectFile",
+                        solutionEx.Message);
```

Comment:
Should we keep the whole exception with stack trace here? (`solutionEx.ToString()`)

Reply:
It would make sense to have the two exception cases do the same thing, I'm not sure which is preferred in this case.

Reply:
It looked unactionable to me that's why I sticked to message
![{6744BFD7-756C-4A18-8B81-8769C8D5EA21}](https://github.com/user-attachments/assets/62638984-6950-49d9-baf8-8d3a24af60b9)

I am not very opinionated, can return callstack :)